src/test/chainstate_write_tests.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2024- The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/setup_common.h>
+#include <validation.h>
+#include <validationinterface.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(chainstate_write_tests)
+
+struct TestSubscriber final : public CValidationInterface {
+    bool m_did_flush{false};
+
+    TestSubscriber() = default;
+
+    void ChainStateFlushed(ChainstateRole role, const CBlockLocator& locator) override
+    {
+        m_did_flush = true;
+    }
+};
+
+BOOST_FIXTURE_TEST_CASE(chainstate_write_interval, TestingSetup)
+{
+    const auto sub{std::make_shared<TestSubscriber>()};
+    m_node.validation_signals->RegisterSharedValidationInterface(sub);
+    Chainstate& chainstate{Assert(m_node.chainman)->ActiveChainstate()};
+    BlockValidationState state_dummy{};
+
+    // Time window to periodically flush is between 50 and 70 minutes
+    SetMockTime(GetTime<std::chrono::minutes>() + 49min);
+    chainstate.FlushStateToDisk(state_dummy, FlushStateMode::PERIODIC);
+    m_node.validation_signals->SyncWithValidationInterfaceQueue();
+    BOOST_CHECK(!sub->m_did_flush);
```

Comment:
The initial run should never flush, right?
`SetMockTime(GetTime<std::chrono::minutes>() + 55min);` passes also, so we should likely update the test
.
Actually, sometimes it passes, other times it fails - this is the reason for my NACK
```
% build/src/test/test_bitcoin --run_test=chainstate_write_tests/chainstate_write_interval        
Running 1 test case...
src/test/chainstate_write_tests.cpp:35: error: in "chainstate_write_tests/chainstate_write_interval": check !sub->m_did_flush has failed

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
% build/src/test/test_bitcoin --run_test=chainstate_write_tests/chainstate_write_interval        
Running 1 test case...

*** No errors detected
% build/src/test/test_bitcoin --run_test=chainstate_write_tests/chainstate_write_interval
Running 1 test case...

*** No errors detected
% build/src/test/test_bitcoin --run_test=chainstate_write_tests/chainstate_write_interval
Running 1 test case...

*** No errors detected
% build/src/test/test_bitcoin --run_test=chainstate_write_tests/chainstate_write_interval
Running 1 test case...

*** No errors detected
% build/src/test/test_bitcoin --run_test=chainstate_write_tests/chainstate_write_interval
Running 1 test case...
src/test/chainstate_write_tests.cpp:35: error: in "chainstate_write_tests/chainstate_write_interval": check !sub->m_did_flush has failed

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
```

Reply:
Well, changing the test to periodically flush after 55 minutes makes it a different test. That would mean the first flush is in the window and could randomly trigger a flush or not.
Was the failure after changing to `55min`?

Reply:
Ahh, I see what you mean. We need to trigger an initial flush to set `m_next_write` first before testing the periodic flush window. Fixed.